test/sort.js

```diff
@@ -21,19 +21,24 @@ const CONFIG_FILES_PRIORITY = [
   'routes.js',
 ];
 
+const getTestSorting = config => (file1, file2, sortRes) => {
+  return [config, cb => cb(file1, file2) === sortRes];
+};
+const testSorting = getTestSorting(CONFIG_FILES_PRIORITY);
+
 metatests.case(
   'Common / sort',
   { common },
   {
     'common.sortComparePriority': [
-      [CONFIG_FILES_PRIORITY, 'files.js', 'sandbox.js', 1],
-      [CONFIG_FILES_PRIORITY, 'filestorage.js', 'routes.js', -1],
-      [CONFIG_FILES_PRIORITY, 'unknown.js', 'sandbox.js', 1],
-      [CONFIG_FILES_PRIORITY, 'log.js', 'sandbox.js', 1],
-      [CONFIG_FILES_PRIORITY, 'sandbox.js', 'sandbox.js', 0],
-      [CONFIG_FILES_PRIORITY, 'log.js', 'log.js', 0],
-      [CONFIG_FILES_PRIORITY, 'tasks.js', 'application.js', -1],
-      [CONFIG_FILES_PRIORITY, 'tasks.js', 'missing_file', -1],
+      testSorting('files.js', 'sandbox.js', 1),
```

Comment:
You should not create or call any functions inside the `metatests.case()` call, as I said in https://github.com/metarhia/common/pull/319#discussion_r362078868, just remove the first element in each test case array, and create the function to test by calling `common.sortComparePriority()` only once before the call to `metatests.case()`.

Reply:
I've found 2 different ways to solve this, but both are not good as for me:

1. By dividing existing `metatests.case()` into 2 parts - with new namespace e.g. `compareFunction`, and another with old `common`.
```
const compareFunction = common.sortComparePriority(CONFIG_FILES_PRIORITY);

metatests.case(
  'Common / sortComparePriority',
  { compareFunction },
  {
    compareFunction: [
      ['files.js', 'sandbox.js', 1],      
      // ...
    ],
  }
);
metatests.case(
  'Common / sort',
  { common },
  {
    'common.sortCompareDirectories': [
      [{ name: '/abc' }, { name: 'abc.ext' }, -1],
      // ...
    ],
  }
);
```

2. By adding a new method to `common` object:
```
common.compareFunction = common.sortComparePriority(CONFIG_FILES_PRIORITY);

metatests.case(
  'Common / sort',
  { common },
  {
    'common.compareFunction': [
      ['files.js', 'sandbox.js', 1],
      ['filestorage.js', 'routes.js', -1],
      // ...
    ],
  }
);
```

Reply:
Just pass the function into the object that is the second argument to the `metatests.case()`:
```js
const compareFunction = common.sortComparePriority(CONFIG_FILES_PRIORITY);

metatests.case(
  'Common / sort',
  { common, sortComparePriority: compareFunction },
  {
    sortComparePriority: [
      ['files.js', 'sandbox.js', 1],
      // ...
    ],
    'common.sortCompareDirectories': [
      [{ name: '/abc' }, { name: 'abc.ext' }, -1],
      // ...
    ],
    // ...
  }
);
```